spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,265 @@
+        reply.setWindowTitle("Spyder")
+        reply.setAttribute(Qt.WA_ShowWithoutActivating)
+        reply.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+        reply.exec_()
+        if reply.result() == QMessageBox.Yes:
+            self.start_installation_update()
+        else:
+            self._change_update_installation_status(status=PENDING)
+
+    def finished_installation(self, status):
+        """Handle finished installation."""
+        if status == FINISHED or status == PENDING:
+            self.setup()
+            self.accept()
+
+    def close_installer(self):
+        """Close the installation dialog."""
+        if (self.status == FINISHED
+                or self.status == CANCELLED):
+            self.setup()
+            self.accept()
+        else:
+            self.hide()
+
+    def reject(self):
+        """Reimplemented Qt method."""
+        on_installation_widget = self._installation_widget.isVisible()
+        if on_installation_widget:
+            self.close_installer()
+        else:
+            super(UpdateInstallerDialog, self).reject()
+
+    def _change_update_installation_status(self, status=NO_STATUS):
+        """Set the installation status."""
+        logger.debug(f"Installation status: {status}")
+        self.status = status
+        self.sig_installation_status.emit(self.status)
+
+    def _progress_reporter(self, block_number, read_size, total_size):
+        progress = 0
+        if total_size > 0:
+            progress = block_number * read_size
+        if self.cancelled:
+            raise UpdateInstallationCancelledException()
+        else:
+            self.sig_download_progress.emit(progress, total_size)
+
+    def cancel_thread_install_update(self):
+        self._change_update_installation_status(status=CANCELLED)
+        self.thread_install_update.join()
+
+    def _download_install(self):
+        try:
+            logger.debug("Downloading installer executable")
+            with TemporaryDirectory(prefix="Spyder-") as tmpdir:
+                destination = os.path.join(tmpdir, 'updateSpyder.exe')
+                url = (
+                    'https://github.com/spyder-ide/spyder/releases/latest/'
+                    'download/Spyder_64bit_lite.exe')
+                is_full_installer = (is_module_installed('numpy') or
+                                     is_module_installed('pandas'))
+                if is_full_installer:
+                    url = (
+                        'https://github.com/spyder-ide/spyder/releases/latest/'
+                        'download/Spyder_64bit_full.exe')
+                logger.debug(f"Downloading installer from: {url}")
+                download = urlretrieve(url,
+                                       destination,
+                                       reporthook=self._progress_reporter)
+                self._change_update_installation_status(status=INSTALLING)
+                install = subprocess.Popen(destination, shell=True)
```

Comment:
```suggestion
                is_full = (is_module_installed('numpy') or
                           is_module_installed('pandas'))
                if os.name == 'nt':
                    name = 'Spyder_64bit_{}.exe'.format('full' if is_full
                                                        else 'lite')
                else:
                    name = 'Spyder{}.dmg'.format('' if is_full else '-Lite')
                url = ('https://github.com/spyder-ide/spyder/releases/latest/'
                       f'download/{name}')
                destination = os.path.join(tmpdir, name)
                logger.debug(f"Downloading installer from: {url}")
                urlretrieve(url, destination,
                            reporthook=self._progress_reporter)
                self._change_update_installation_status(status=INSTALLING)

                cmd = ('' if os.name == 'nt' else 'open ') + destination
                install = subprocess.Popen(cmd, shell=True)
```

This should handle macOS as well.